### ISSUE TITLE
BUG: internal linter error: `'CommentedMap' object has no attribute 'split'`

### ISSUE THREAD

**Issue by h-vetinari** [conda-forge/conda-smithy#2165]
As of https://github.com/conda-forge/torchaudio-feedstock/pull/6/commits/488d67862e45edd58b684014a84b147c0ae8d276 in https://github.com/conda-forge/torchaudio-feedstock/pull/6, the linter fails with
```
>conda smithy lint --conda-forge recipe
Traceback (most recent call last):
  File "E:\miniforge\envs\builder\Scripts\conda-smithy-script.py", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "E:\miniforge\envs\builder\Lib\site-packages\conda_smithy\cli.py", line 767, in main
    args.subcommand_func(args)
  File "E:\miniforge\envs\builder\Lib\site-packages\conda_smithy\cli.py", line 645, in __call__
    lints, hints = linter.main(
                   ^^^^^^^^^^^^
  File "E:\miniforge\envs\builder\Lib\site-packages\conda_smithy\lint_recipe.py", line 697, in main
    results, hints = lintify_meta_yaml(
                     ^^^^^^^^^^^^^^^^^^
  File "E:\miniforge\envs\builder\Lib\site-packages\conda_smithy\lint_recipe.py", line 235, in lintify_meta_yaml
    run_conda_forge_specific(
  File "E:\miniforge\envs\builder\Lib\site-packages\conda_smithy\lint_recipe.py", line 544, in run_conda_forge_specific
    dep = rq.split(" ")[0].strip()
          ^^^^^^^^
AttributeError: 'CommentedMap' object has no attribute 'split'
```

**Comment by h-vetinari** [comment]
@adament had been commenting on zulip about a closely-related issue. I've just now tested an install from main (thus including https://github.com/conda-forge/conda-smithy/pull/2157), but the issue still remains.

**Comment by wolfv** [comment]
@h-vetinari do you need help fixing this?

**Comment by wolfv** [comment]
Fixed in #2166

**Comment by h-vetinari** [comment]
Fix confirmed! :) thank you very much! 🙏

**Comment by h-vetinari** [comment]
Still running into this with smithy 3.44.9 (despite the fix from #2166 which landed in 3.44.8) in https://github.com/conda-forge/torchaudio-feedstock/pull/12
```
(builder) E:\conda-forge\torchaudio-feedstock>conda smithy lint recipe --conda-forge
Traceback (most recent call last):
  File "E:\miniforge\envs\builder\Scripts\conda-smithy-script.py", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "E:\miniforge\envs\builder\Lib\site-packages\conda_smithy\cli.py", line 767, in main
    args.subcommand_func(args)
  File "E:\miniforge\envs\builder\Lib\site-packages\conda_smithy\cli.py", line 645, in __call__
    lints, hints = linter.main(
                   ^^^^^^^^^^^^
  File "E:\miniforge\envs\builder\Lib\site-packages\conda_smithy\lint_recipe.py", line 727, in main
    results, hints = lintify_meta_yaml(
                     ^^^^^^^^^^^^^^^^^^
  File "E:\miniforge\envs\builder\Lib\site-packages\conda_smithy\lint_recipe.py", line 235, in lintify_meta_yaml
    run_conda_forge_specific(
  File "E:\miniforge\envs\builder\Lib\site-packages\conda_smithy\lint_recipe.py", line 574, in run_conda_forge_specific
    dep = rq.split(" ")[0].strip()
          ^^^^^^^^
AttributeError: 'CommentedMap' object has no attribute 'split'
```

**Comment by mgorny** [comment]
Just hit the same (?) error while working on torchvision-feedstock port. FWICS the problem happens when conditions are nested, e.g.:

https://github.com/conda-forge/torchvision-feedstock/blob/39add14ded9c979a012a643c2b6fa7c2230177f6/recipe/recipe.yaml#L113-L120

At the moment of failure, `rq` is:

```python
{'if': 'cuda_compiler_version != "None"', 'then': ['libcublas-dev', 'libcusolver-dev', 'libcusparse-dev', 'libnvjpeg-dev']}
```
